packages/ember/tests/component_registration_test.js

```diff
@@ -1,200 +1,172 @@
 import { Controller } from 'ember-runtime';
-import { run } from 'ember-metal';
-
-import { Application } from 'ember-application';
-import { Router } from 'ember-routing';
-import { compile } from 'ember-template-compiler';
-import {
-  Component,
-  setTemplates,
-  setTemplate
-} from 'ember-glimmer';
-import { jQuery } from 'ember-views';
-
-let App, appInstance;
-
-function prepare() {
-  setTemplate('components/expand-it', compile('<p>hello {{yield}}</p>'));
-  setTemplate('application', compile('Hello world {{#expand-it}}world{{/expand-it}}'));
-}
-
-function cleanup() {
-  run(() => {
-    try {
-      if (App) {
-        App.destroy();
-      }
-      App = appInstance = null;
-    } finally {
-      setTemplates({});
-    }
-  });
-}
-
-QUnit.module('Application Lifecycle - Component Registration', {
-  setup: prepare,
-  teardown: cleanup
-});
-
-function boot(callback, startURL = '/') {
-  run(() => {
-    App = Application.create({
-      name: 'App',
-      rootElement: '#qunit-fixture'
-    });
+import { Component } from 'ember-glimmer';
+import { moduleFor, ApplicationTestCase } from 'internal-test-helpers';
 
-    App.deferReadiness();
+moduleFor('Application Lifecycle - Component Registration', class extends ApplicationTestCase {
 
-    App.Router = Router.extend({
-      location: 'none'
+  ['@test The helper becomes the body of the component'](assert) {
+    this.addTemplate('application', 'Hello world {{#expand-it}}world{{/expand-it}}');
+    this.addComponent('expand-it', {
+      template: '<p>hello {{yield}}</p>'
```

Comment:
where is `registerComponent` defined? I'm not sure what you're looking at, but I expect `registerComponent` would register the factory into the registry. There are lots of tests in the Ember suite that tried to avoid the global default resolver by registering all their factories into the registry. The problem is that this doesn't really reflect reality, where the resolver almost always provides a user-level factory into the container.

Adding the `TestResolver` is our way around the default resolver while also avoiding the registry hack used in some Glimmer tests. Eventually it would be nice to update tests using the registry, but it isn't a priority.

`addComponent` here doesn't seem necessary. It is just a convenience method for adding both a template and a component class to the resolver at the same time. You could safely rewrite this as:

```js
this.addTeamplate('component:expand-it', `<p>hello {{yield}}</p>`);
```